[Debugger] `llvm-kompile-testing` doesn't generate debug symbols correctly

The `llvm-kompile-testing` doesn't generate the debug symbols for a k file as expected.

- OS:
```bash
lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.3 LTS
Release:        22.04
Codename:       jammy
```
- K Version:
```
K version:    v6.0.144-0-g0cfe674f6f
Build date:   Sun Oct 15 15:11:21 -03 2023
```
- LLVM Backend Last Commit:
```bash
git log | head -n 1
commit a859fb6d77a48220d9d8b9314371933e8204c1d3
```

## Expected Behavior:
```bash
llvm-kompile-testing definition.kore main -g -o interpreter
gdb --args ./interpreter input -1 output
```
```bash
GNU gdb (Ubuntu 12.1-0ubuntu1~22.04) 12.1
Copyright (C) 2022 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from ./simple-kompiled/interpreter...
(gdb) k start
Temporary breakpoint 1 at 0x640a0
Starting program: /home/robertorosmaninho/rv/tests/issue_840/simple-kompiled/interpreter input -1 output
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".

Temporary breakpoint 1, 0x00005555555b80a0 in main ()
0x000055555559ccd0 in k_step (subject=<k>
  0 ~> .
</k>) at /home/robertorosmaninho/rv/tests/issue_840/simple.k:4
4         rule I => I +Int 1
(gdb) q
A debugging session is active.

        Inferior 1 [process 76700] will be killed.

Quit anyway? (y or n) y
```
This behavior was achieved using an `interpreter` generated by `kompile`.

## Actual Behavior
```bash
lvm-kompile-testing definition.kore main -g -o interpreter
gdb --args ./interpreter input -1 output 
```
```bash               
GNU gdb (Ubuntu 12.1-0ubuntu1~22.04) 12.1
Copyright (C) 2022 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from ./interpreter...
(gdb) k start
Temporary breakpoint 1 at 0x6a690
Starting program: /home/robertorosmaninho/rv/tests/issue_840/interpreter input -1 output
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".

Temporary breakpoint 1, 0x00005555555be690 in main ()
Python Exception <class 'gdb.error'>: No source file named definition.kore.
Error occurred in Python: No source file named definition.kore.
(gdb) q
A debugging session is active.

        Inferior 1 [process 77008] will be killed.

Quit anyway? (y or n) y
```

## Steps to reproduce:
- K File: simple.k
```k
module SIMPLE
  imports INT

  rule I => I +Int 1
    requires I <Int 100
endmodule
```
  - `input` as:
  ```kore
  LblinitGeneratedTopCell{}(Lbl'Unds'Map'Unds'{}(Lbl'Stop'Map{}(),Lbl'UndsPipe'-'-GT-Unds'{}(inj{SortKConfigVar{}, SortKItem{}}(\dv{SortKConfigVar{}}("$PGM")),inj{SortInt{}, SortKItem{}}(\dv{SortInt{}}("0")))))
  ```
  - `output` as:
  ```kore
Lbl'-LT-'generatedTop'-GT-'{}(Lbl'-LT-'k'-GT-'{}(kseq{}(inj{SortInt{}, SortKItem{}}(\dv{SortInt{}}("100")),dotk{}())),Lbl'-LT-'generatedCounter'-GT-'{}(\dv{SortInt{}}("0")))Lbl'-LT-'generatedTop'-GT-'{}(Lbl'-LT-'k'-GT-'{}(kseq{}(inj{SortInt{}, SortKItem{}}(\dv{SortInt{}}("100")),dotk{}())),Lbl'-LT-'generatedCounter'-GT-'{}(\dv{SortInt{}}("0")))
  ```

- Kompile: (To generate the `definition.kore`)
```bash
kompile simple.k --syntax-module SIMPLE --backend llvm --enable-llvm-debug
```
- Generate `interpreter` from `llvm-kompile-testing`:
```bash
llvm-kompile-testing definition.kore main -g -o interpreter
```
- Test GDB with llvm-kompile-testing's interpreters:
```bash
gdb --args ./interpreter input -1 output
k start
```
- Check expected behavior is kompiler's interpreter:
```bash
gdb --args ./simple-kompiled/interpreter input -1 output
k start
```

I'm having issues reproducing this with the latest versions of `llvm-backend`:

### LLVM Backend
```
git log | head -n 1
commit 29bb01e45da52e13549f4f9a38dac91aacb6d0bd
```

### K
```
kompile --version
K version:    v6.1.12-0-gda72fc4726-dirty
Build date:   Tue Nov 14 11:31:32 -03 2023
```

### Reproduction
```
kompile simple.k --syntax-module SIMPLE --backend llvm --enable-llvm-debug
llvm-kompile-testing /home/robertorosmaninho/rv/tests/issue_840/simple-kompiled/definition.kore main -g -o interpreter
```

### Error:
```
llvm-kompile-testing /home/robertorosmaninho/rv/tests/issue_840/simple-kompiled/definition.kore main -g -o interpreter
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.6.0:java (default-cli) on project llvm-backend-matching: An exception occured while executing the Java class. ERROR: Line 307: Column 26: Expected <Identifier>, but '\'
[ERROR]   axiom{} \right-assoc{}(\or{SortKItem{}} (\exists{SortKItem{}} (Val:SortList{}, inj{SortList{}, SortKItem{}} (Val:SortList{})), \exists{SortKItem{}} (Val:SortKConfigVar{}, inj{SortKConfigVar{}, SortKItem{}} (Val:SortKConfigVar{})), \exists{SortKItem{}} (Val:SortGeneratedCounterCellOpt{}, inj{SortGeneratedCounterCellOpt{}, SortKItem{}} (Val:SortGeneratedCounterCellOpt{})), \exists{SortKItem{}} (Val:SortGeneratedCounterCell{}, inj{SortGeneratedCounterCell{}, SortKItem{}} (Val:SortGeneratedCounterCell{})), \exists{SortKItem{}} (Val:SortBool{}, inj{SortBool{}, SortKItem{}} (Val:SortBool{})), \exists{SortKItem{}} (Val:SortKCell{}, inj{SortKCell{}, SortKItem{}} (Val:SortKCell{})), \exists{SortKItem{}} (Val:SortMap{}, inj{SortMap{}, SortKItem{}} (Val:SortMap{})), \exists{SortKItem{}} (Val:SortKCellOpt{}, inj{SortKCellOpt{}, SortKItem{}} (Val:SortKCellOpt{})), \exists{SortKItem{}} (Val:SortInt{}, inj{SortInt{}, SortKItem{}} (Val:SortInt{})), \exists{SortKItem{}} (Val:SortGeneratedTopCell{}, inj{SortGeneratedTopCell{}, SortKItem{}} (Val:SortGeneratedTopCell{})), \exists{SortKItem{}} (Val:SortSet{}, inj{SortSet{}, SortKItem{}} (Val:SortSet{})), \exists{SortKItem{}} (Val:SortGeneratedTopCellFragment{}, inj{SortGeneratedTopCellFragment{}, SortKItem{}} (Val:SortGeneratedTopCellFragment{})), \bottom{SortKItem{}}())) [constructor{}()] // no junk
[ERROR]                          ^
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

--- a/bin/llvm-kompile-testing
+++ b/bin/llvm-kompile-testing
@@ -12,4 +12,4 @@ definition=$(realpath "$1")
 mode="$2"
 shift; shift
 ( cd "@PROJECT_SOURCE_DIR@"/matching && mvn exec:java -Dexec.args="$definition qbaL $dt_dir 1" -q )
-llvm-kompile "$definition" "$dt_dir" "$mode" "$@" -g
+llvm-kompile "$definition" "$dt_dir" "$mode" -- "$@" -g

--- a/bin/llvm-kompile-testing
+++ b/bin/llvm-kompile-testing
@@ -2,7 +2,7 @@
 set -e
 
 if [ $# -lt 2 ]; then
-  echo "Usage: $0 <definition.kore> [main|library|search|static|python|pythonast|c] <clang flags>"
+  echo "Usage: $0 <definition.kore> [main|library|search|static|python|pythonast|c] <llvm-kompile flags> [--] <clang flags>"
   echo "See llvm-kompile -h for help"
   exit 1
 fi
@@ -12,4 +12,23 @@ definition=$(realpath "$1")
 mode="$2"
 shift; shift
 ( cd "@PROJECT_SOURCE_DIR@"/matching && mvn exec:java -Dexec.args="$definition qbaL $dt_dir 1" -q )
-llvm-kompile "$definition" "$dt_dir" "$mode" -- "$@" -g
+
+llvm_kompile_flags=()
+clang_flags=()
+
+while [ $# -gt 0 ]; do
+  if [ "$1" = "--" ]; then
+    clang_flags+=("--")
+    shift
+    break
+  fi
+  llvm_kompile_flags+=("$1")
+  shift
+done
+
+while [ $# -gt 0 ]; do
+  clang_flags+=("$1")
+  shift
+done
+
+llvm-kompile "$definition" "$dt_dir" "$mode" "${llvm_kompile_flags[@]}" "${clang_flags[@]}" -g

